server/src/main/java/org/elasticsearch/common/bytes/AbstractBytesReference.java

```diff
@@ -31,6 +31,14 @@ public int getIntLE(int index) {
         return (get(index + 3) & 0xFF) << 24 | (get(index + 2) & 0xFF) << 16 | (get(index + 1) & 0xFF) << 8 | get(index) & 0xFF;
     }
 
+    @Override
+    public double getDoubleLE(int index) {
+        long bits = (long) (get(index + 7) & 0xFF) << 56 | (long) (get(index + 6) & 0xFF) << 48 | (long) (get(index + 5) & 0xFF) << 40
+            | (long) (get(index + 4) & 0xFF) << 32 | (long) (get(index + 3) & 0xFF) << 24 | (get(index + 2) & 0xFF) << 16 | (get(index + 1)
+                & 0xFF) << 8 | get(index) & 0xFF;
```

Comment:
I don't know if we want to do it in this PR or in a follow up, but we'll want this same bit twiddling for the `long` version, and might as well make this reusable for that case.

Reply:
👍 I will add `getLongLE` method to this class and the interface and a unit test for it.

Reply:
pushed: 94c75f3cf9d99f310648445eb5204788bb0e99d7